### DISCUSSION TITLE
Live chat icon

### DISCUSSION THREAD

**Discussion by ahmedabbous** [TDRA-ae/aegov-dls#31] [General]
Hello
the live chat usually provided by third party where we can't control its icon or even its position, and there is no way for workaround to manipulate it using JS, is it ok not follow the DLS design in this case?

**Comment by moiz-conceptlz** [comment]
Hello @ahmedabbous 

This is quite true; you will most likely use a chat provided by a 3rd party.

However, understanding the trigger element of the chat, can perhaps help understand the HTML structure to allow customisation. CSS can always be used to override the style that may be added as a default. If the 3rd party documentation allows JS triggers as custom triggers, it is strongly recommended to do that. 

Every 3rd Party chat system comes with a JS code that pre-adds a block of HTML/CSS and JS that creates the icon to activate a chat. Many of such software allow you backend access to change the icon and customise the colors, icon, style etc using a interactive method. Many also allow custom CSS additions. 

However, nothing stops you from using a barebone HTML/CSS or overriding the CSS preloaded by the chat software. 

The design system recommends the use of floating icons, as mentioned on [https://designsystem.gov.ae/guidelines/actions](https://designsystem.gov.ae/guidelines/actions) where the purpose is - to now allow floating icons to disrupt the view of content or actions. Your floating icon - chat or anything else - must be managed well for mobile.
